craft_application/errors.py

```diff
@@ -141,8 +142,12 @@ def __init__(self) -> None:
 class MultipleBuildsError(CraftError):
     """The build plan contains multiple possible builds."""
 
-    def __init__(self) -> None:
-        message = "Multiple builds match the current platform."
+    def __init__(self, matching_builds: list[models.BuildInfo] | None = None) -> None:
+        message = "Multiple builds match the current platform"
+        if not matching_builds:
+            message += "."
+        else:
+            message += ": " + ", ".join([build.platform for build in matching_builds])
```

Comment:
Don't forget you can do `utils.humanize_list([build.platform for build in matching_builds], "and")`

Just a suggestion, I don't have a preference

Reply:
I wasn't aware of `humanize_list`, going to make use of it